react-hash-routing/client/app.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import Header from './components/header';
+import Catalog from './pages/catalog';
+import NotFound from './pages/not-found';
+import ProductDetails from './pages/product-details';
+import { parseRoute } from './lib';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      route: parseRoute(window.location.hash)
+    };
+  }
+
+  componentDidMount() {
+    /**
+     * Listen for hash change events on the window object
+     * Each time the window.location.hash changes, parse
+     * it with the parseRoute() function and update state
+     */
+    const route = parseRoute(window.location.hash);
+    this.setState({ route: route });
```

Comment:
You need an event listener here. You should add an event listener to the window listening for a `hashchange` event